relay-server/src/metrics_extraction/event.rs

```diff
@@ -45,22 +76,39 @@ pub fn extract_event_metrics(event: &Event, config: &MetricExtractionConfig) ->
             continue;
         };
 
-        let Some(value) = read_metric_value(event, metric_spec.field.as_deref(), mri.ty) else {
+        let Some(value) = read_metric_value(data, metric_spec.field.as_deref(), mri.ty) else {
             continue;
         };
 
+        // Combine global tag mapping with metric's own tags.
+        // Global tags are overwritten by metric-specific tags.
+        let tags = config
+            .tags
+            .iter()
+            .filter_map(|t| {
+                // TODO: support wildcards as the docs suggest.
+                t.metrics
+                    .contains(&metric_spec.mri)
+                    .then_some(t.tags.iter())
+            })
+            .flatten()
+            .chain(metric_spec.tags.iter());
```

Comment:
Previous implementation seemed to ignore the `MetricTagMapping`. Added it here, but without wildcard support.